extensions/amp-story-shopping/0.1/amp-story-shopping-config.js

```diff
@@ -8,23 +16,99 @@ export class AmpStoryShoppingConfig extends AMP.BaseElement {
     super(element);
 
     /** @private {string} */
-    this.myText_ = TAG;
+    this.element_ = element;
+
+    /** @private @const {!./amp-story-store-service.AmpStoryStoreService} */
+    this.storeService_ = null;
+  }
 
-    /** @private {?Element} */
-    this.container_ = null;
+  /**
+   * Keys product data to product-ids and adds them to the store service.
+   * @private
+   * @param {!JsonObject} storyConfig
+   */
+  dispatchConfig_(storyConfig) {
+    const productIDtoProduct = {};
+
+    for (const item of storyConfig['items']) {
+      productIDtoProduct[item['product-tag-id']] = item;
+    }
+
+    this.storeService_.dispatch(Action.ADD_SHOPPING_STATE, productIDtoProduct);
+
+    //TODO: add call to validate config here. See validation I2I #36412 for implementation details
   }
 
   /** @override */
   buildCallback() {
     super.buildCallback();
-    this.container_ = this.element.ownerDocument.createElement('div');
-    this.container_.textContent = this.myText_;
-    this.element.appendChild(this.container_);
-    applyFillContent(this.container_, /* replacedContent */ true);
+    return Services.storyStoreServiceForOrNull(this.win).then(
+      (storeService) => {
+        devAssert(storeService, 'Could not retrieve AmpStoryStoreService');
+
+        this.storeService_ = storeService;
+
+        this.getConfig().then((storyConfig) => {
+          //if remote config is not valid, try using the inline config
+          if (storyConfig == null && this.element_.hasAttribute('src')) {
+            this.getInlineConfig_(this.element_.firstElementChild).then(
+              (storyInlineconfig) => {
+                this.dispatchConfig_(storyInlineconfig);
+              }
+            );
+          } else {
+            this.dispatchConfig_(storyConfig);
+          }
+        });
```

Comment:
Since we're using promises, we can remove the `else if` and use `catch`. 
This could be refactored to something like:
```
this.getConfig()
  .then((storyConfig) => this.dispatchConfig_(storyConfig))
  .catch(() => {
    this.getInlineConfig_().then((storyInlineconfig) =>
      this.dispatchConfig_(storyInlineconfig)
    );
  });
```

There might be more opportunities to save bytes.